airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/array.sql

```diff
@@ -25,7 +25,7 @@
         case jsonb_typeof({{ array_col }})
         when 'array' then {{ array_col }}
         else '[]' end
-    ) as {{ array_col }}
```

Comment:
Following discussion on slack: https://airbytehq.slack.com/archives/C01MFR03D5W/p1619129248210400

This is getting translated on postgres into:
```
select
    _airbyte_bitrix_deal_report_hashid,
    jsonb_extract_path_text(user_fields, 'UF_ID') as uf_id,
    jsonb_extract_path_text(user_fields, 'UF_VALUE') as uf_value,
    _airbyte_emitted_at
from "data".public."bitrix_deal_report"
cross join jsonb_array_elements(
        case jsonb_typeof(user_fields)
        when 'array' then user_fields
        else '[]' end
    ) as user_fields
where user_fields is not null
-- user_fields at bitrix_deal_report/USER_FIELDS
```

which doesn't work... but if it's named as:
```
select
    _airbyte_bitrix_deal_report_hashid,
    jsonb_extract_path_text(user_fields_unnested, 'UF_ID') as uf_id,
    jsonb_extract_path_text(user_fields_unnested, 'UF_VALUE') as uf_value,
    _airbyte_emitted_at
from public."bitrix_deal_report"
cross join jsonb_array_elements(
        case jsonb_typeof(user_fields)
        when 'array' then user_fields
        else '[]' end
    ) as user_fields_unnested
where user_fields_unnested is not null
-- user_fields at bitrix_deal_report/USER_FIELDS
```

Then it's good!
Here I just call the intermediate column `_airbyte_nested_data`

Reply:
to make sure I understand, this is happening because in the original query, the names are identical so it's actually trying to reference the underlying data instead of the unnested one?

Reply:
> to make sure I understand, this is happening because in the original query, the names are identical so it's actually trying to reference the underlying data instead of the unnested one?

Yes, referring to `user_fields` is ambiguous and the json_extract_path_text in postgres does not seem to behave as expected (other destinations seems to be handling that ambiguity just fine based on the context)